### DISCUSSION TITLE
Security http - get target path from multiple firewall with same context?

### DISCUSSION THREAD

**Discussion by lugosium** [symfony/symfony#54793] [Q&A]
Hi,

I have an issue where after login i am not redirected to requested page. 
After some debugging i found out that it's because the requested page belong to another firewall than the login page. (https://github.com/symfony/security-http/blob/7.0/Util/TargetPathTrait.php#L36)

I looked at the code and it seems impossible to allow this behavior?

And as the code is in the `security-http` library, i'm not sure it is possible to create a MR as we cannot retrieve firewalls with same context from here?

Thanks!

**Comment by MatTheCat** [comment]
Hello,

from your issue it sounds like the `TargetPathTrait` should use the firewall’s context rather than its name.

That would require changing the definition of the services using this trait.
